pgmpy/inference/base_continuous.py

```diff
@@ -0,0 +1,268 @@
+"""
+    A collection of base class objects for continuous model
+"""
+
+import numpy as np
+
+
+class JointGaussianDistribution(object):
+    """
+    A naive gaussian container class.(Depricated)
+    Will be removed when Multivariate Distributions will be implemented
+
+    Paramters
+    ---------
+    mean_vec: A vector (row matrix or a column matrix or a 1d array type structure)
+              Represents the mean of the distribution
+              Will be converted into a column matrix of appropriate shape
+
+    cov_matrix: A numpy.array or numpy.matrix of size len(mean_vec) x len(mean_vec),
+                Covariance matrix for the distribution.
+
+    """
+
+    def __init__(self, mean_vec, cov_matrix):
+
+        if isinstance(mean_vec, (np.matrix, np.ndarray, list, tuple, set, frozenset)):
+            mean_vec = np.array(mean_vec).flatten()
+        else:
+            raise TypeError("mean_vec should be a 1d array type object")
+        mean_vec = np.reshape(mean_vec, (len(mean_vec), 1))
+
+        if not isinstance(cov_matrix, (np.matrix, np.ndarray, list)):
+            raise TypeError(
+                "cov_matrix must be numpy.matrix type object")
+
+        if cov_matrix.shape[0] != cov_matrix.shape[1]:
+            raise ValueError(
+                "cov_matrix must be a square matrix")
+
+        if mean_vec.shape[0] != cov_matrix.shape[0]:
+            raise ValueError("shape of mean vector should be d X 1 and" +
+                             " shape of covariance matrix should be d X d")
+
+        self.mean_vec = mean_vec
+        self.cov_matrix = np.array(cov_matrix)
+        self.precision_matrix = np.linalg.inv(cov_matrix)
+
+
+class BaseGradLogPDF(object):
```

Comment:
@khalibartan Do you think gradient computation should be a method in `ContinuousFactor` or classes inheriting it ?